src/System.Windows.Extensions/tests/System/Drawing/ImageFormatConverterTests.cs

```diff
@@ -0,0 +1,227 @@
+            Assert.False(_imgFmtConvFrmTD.CanConvertTo(null, typeof(object)), "TD object");
+            Assert.False(_imgFmtConvFrmTD.CanConvertTo(null, typeof(int)), "TD int");
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void TestConvertFrom_ImageFormatToString()
+        {
+            Assert.Equal(_imageFmt, (ImageFormat)_imgFmtConv.ConvertFrom(null, CultureInfo.InvariantCulture, ImageFormat.Bmp.ToString()));
+            Assert.Equal(_imageFmt, (ImageFormat)_imgFmtConvFrmTD.ConvertFrom(null, CultureInfo.InvariantCulture, ImageFormat.Bmp.ToString()));
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void TestConvertFrom_ThrowsNotSupportedException()
+        {
+            Assert.Throws<NotSupportedException>(() => _imgFmtConv.ConvertFrom(null, CultureInfo.InvariantCulture, ImageFormat.Bmp));
+            Assert.Throws<NotSupportedException>(() => _imgFmtConv.ConvertFrom(null, CultureInfo.InvariantCulture, ImageFormat.Bmp.Guid));
+            Assert.Throws<NotSupportedException>(() => _imgFmtConv.ConvertFrom(null, CultureInfo.InvariantCulture, new object()));
+            Assert.Throws<NotSupportedException>(() => _imgFmtConv.ConvertFrom(null, CultureInfo.InvariantCulture, 10));
+                          
+            Assert.Throws<NotSupportedException>(() => _imgFmtConvFrmTD.ConvertFrom(null, CultureInfo.InvariantCulture, ImageFormat.Bmp));
+            Assert.Throws<NotSupportedException>(() => _imgFmtConvFrmTD.ConvertFrom(null, CultureInfo.InvariantCulture, ImageFormat.Bmp.Guid));
+            Assert.Throws<NotSupportedException>(() => _imgFmtConvFrmTD.ConvertFrom(null, CultureInfo.InvariantCulture, new object()));
+            Assert.Throws<NotSupportedException>(() => _imgFmtConvFrmTD.ConvertFrom(null, CultureInfo.InvariantCulture, 10));
+        }
+
+        private ImageFormat ConvertFromName(string imgFormatName)
+        {
+            return (ImageFormat)_imgFmtConvFrmTD.ConvertFrom(null, CultureInfo.InvariantCulture, imgFormatName);
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void ConvertFrom_ShortName()
+        {
+            Assert.Equal(ImageFormat.Bmp, ConvertFromName("Bmp"));
+            Assert.Equal(ImageFormat.Emf, ConvertFromName("Emf"));
+            Assert.Equal(ImageFormat.Exif, ConvertFromName("Exif"));
+            Assert.Equal(ImageFormat.Gif, ConvertFromName("Gif"));
+            Assert.Equal(ImageFormat.Tiff, ConvertFromName("Tiff"));
+            Assert.Equal(ImageFormat.Png, ConvertFromName("Png"));
+            Assert.Equal(ImageFormat.MemoryBmp, ConvertFromName("MemoryBmp"));
+            Assert.Equal(ImageFormat.Icon, ConvertFromName("Icon"));
+            Assert.Equal(ImageFormat.Jpeg, ConvertFromName("Jpeg"));
+            Assert.Equal(ImageFormat.Wmf, ConvertFromName("Wmf"));
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void ConvertFrom_LongName()
```

Comment:
This test should be skipped on NETFX since we're adding the support to convert from long name in .NET Core and it is not in .NET Framework. You can add attribute to do so:

```cs
[SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Suppor to convert image format from long name was added to .NET Core directly")]
```